internal/databases/mongo/binary/mongod.go

```diff
@@ -105,7 +106,36 @@ func (mongodService *MongodService) GetBackupCursor() (cursor *mongo.Cursor, err
 			time.Sleep(time.Minute * minutes)
 		}
 	}
-	return cursor, err
+
+	backupContext, stopPingBackupCursor := context.WithCancel(mongodService.Context)
+	var closeBackupCursorWg sync.WaitGroup
```

Comment:
Maybe waitgroup is overkill here? we can just use a simple channel instead.

Reply:
I used channel to notify closing before waitgroup, but I don't "wait" closing backup cursor and I think that backup cursor don't close. Maybe problem was in other places. Here I wait closing backup cursor explicitly